### DISCUSSION TITLE
How often does L1 update the timestamp to L2?

### DISCUSSION THREAD

**Discussion by KittyFu307** [zkSync-Community-Hub/zksync-developers#24] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

Our project involves the application of timestamps; so we need to know how often will zkSyncEra
Synchronize the timestamp of L1 to L2; here is an example, the user involves `requestWithdraw` and `executeWithdraw` when withdrawing token, if the `pendingTime` is too small, (if it is 1s or 5s, the user manually count), it is possible to make `requestWithdraw` and `executeWithdraw` at the same time timestamp;

### What is the specific issue or error you're encountering?

no

### Can you share the error messages or logs you're receiving, if any?

no

### Have you made any recent changes to the contract before encountering this issue?

no

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

no

### Have you tried to isolate the problem, and if so, what were the results?

no

### What steps have you already taken to try to resolve the issue?

I have already known the batch time is happened in here:
https://etherscan.io/address/0x32400084c286cf3e17e7b677ea9583e60a000324
and the update method is `requestL2Transaction `
I also check the doc and find that https://era.zksync.io/docs/api/api.html#zks-getl1batchblockrange 
Is that when the `blockrange` changes, the timestamp will be updated?

But I do not know the frequency of this happened. Could you guys give me some more info on that? Or is there any other API that I can check to redesign our system? Thank you so much

**Comment by uF4No** [comment, marked as answer]
To understand how `block.timestamp` first you need to know about different types of blocks in Era: L2 blocks and batches. You can [find more info in this page of the docs](https://era.zksync.io/docs/reference/concepts/transactions/blocks.html#blocks-in-zksync-era).

As mentioned in [this discussion](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/13), currently `block.timestamp` returns the timestamp from the latest produced batch. The frequency to produce batches is not fixed and involves multiple factors [mentioned in the blocks page in the docs](https://era.zksync.io/docs/reference/concepts/transactions/blocks.html#processing-times). In general, batches will be produced faster when there's more usage in the network. 

With the current settings and load on mainnet the protocol is generating batches every 30-60secs.

We're working on an update that will give developers the option to retrieve the block timestamp from both the batches and the L2 blocks.

**Comment by KittyFu307** [comment]
I can assume the updating of block.timestamp every 30-60s
I also check the doc and found the batch timeout of the processing time. Is the batch timeout still 24 hours?

**Reply by uF4No** [reply]
The batch timeout is currently one hour but it's a setting that can be adjusted as well

**Reply by KittyFu307** [reply]
Sounds great! thanks

**Comment by steyn52** [comment]
In zkSync, the synchronization of timestamps from L1 to L2 occurs with every batch processing cycle. The frequency of this synchronization depends on the batch time configured for the zkSync network. You can monitor the batch block range using the getL1BatchBlockRange API to track when the timestamp is updated. If you need more precise control over timestamp synchronization, you may consider adjusting your application logic to accommodate the batch processing cycle of zkSync.

**Comment by cameronstroman** [comment]
Functionality: Understanding the synchronization frequency of timestamps between L1 and L2 on zkSync Era Mainnet.
Issue: Uncertain about the frequency of timestamp synchronization between L1 and L2.

**Comment by unticcutecerin** [comment]
The user is asking how often zkSyncEra synchronizes the timestamp of L1 to L2 and how they can redesign their system to account for this synchronization. The user has identified the requestL2Transaction method and the zks_getL1BatchBlockRange API but is unsure of the frequency of updates. Without further information or access to the code repository, it is difficult to provide a specific solution. However, it may be helpful for the user to consult the zkSync documentation or reach out to the zkSync support team for guidance on how to design their system to account for timestamp synchronization and obtain information on the frequency of updates.
